### ISSUE TITLE
DNSSEC / EDNS / Transfer Support

### ISSUE THREAD

**Issue by echodreamz** [TechnitiumSoftware/DnsServer#97]
This server has been ROCK SOLID, really happy with it, great work. Just checking to see if DNSSEC signed zones, EDNS support and possibly zone transfer to/from other DNS servers is on the roadmap for the future?

Thanks!

**Comment by ShreyasZare** [comment]
Thanks for the compliments. Yes, these features are planned for future releases. Next release will support running primary and secondary servers using zone transfer. Later releases will include EDNS and DNSSEC. DNSSEC will take time to implement and thus no time can be committed for its release.

**Comment by ShreyasZare** [comment]
Zone transfer support is now available but, is limited and only available between primary and secondary name servers.

**Comment by thb007** [comment]
> Zone transfer

Hello author, is there any example of how to use Zone transfer？

**Comment by ShreyasZare** [comment]
> 
> 
> > Zone transfer
> 
> Hello author, is there any example of how to use Zone transfer？

Zone transfer is supported between primary and secondary servers. Secondary servers are by default allowed to do zone transfer. You can do AXFR queries from the DNS client if the primary zone is hosted locally for testing purposes. Let me know if you have any more queries.

**Comment by thb007** [comment]
> > > Zone transfer
> > 
> > 
> > Hello author, is there any example of how to use Zone transfer？
> 
> Zone transfer is supported between primary and secondary servers. Secondary servers are by default allowed to do zone transfer. You can do AXFR queries from the DNS client if the primary zone is hosted locally for testing purposes. Let me know if you have any more queries.

Both of my two servers have TechnitiumSoftware/DnsServer installed, so how do I configure to perform zone transfer?

**Comment by ShreyasZare** [comment]
> Both of my two servers have TechnitiumSoftware/DnsServer installed, so how do I configure to perform zone transfer?

You can create a primary zone on one server and then create secondary zone on the another one. You need to add NS record in the primary zone that points to the domain name of the secondary server. 

If the secondary server's domain name is not directly resolvable then you will need to add its IP address as glue addresses for the NS record.

If the primary zone is not directly resolvable then you will need to specify the first server's IP address while adding the secondary zone. 

Once the secondary zone is added, it will automatically do zone transfer to sync all records from the primary server. Any change you make in the primary zone will thus get synced automatically to the secondary zone within a few seconds.

**Comment by SunRed** [comment]
Are there any plans on supporting at least DNSSEC signature validation for recursive queries already?

**Comment by ShreyasZare** [comment]
> 
> 
> Are there any plans on supporting at least DNSSEC signature validation for recursive queries already?

Yes, DNSSEC is planned but its going to take some time as there are other features prioritized for implementation.

**Comment by ShreyasZare** [comment]
New version 6.3 now supports configuring zone transfer and notify options.

**Comment by thb007** [comment]
> New version 6.3 now supports configuring zone transfer and notify options.

thanks

**Comment by ShreyasZare** [comment]
Technitium DNS Server v8 is now released which EDNS and DNSSEC. Do check and let me know your feedback.
